chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -482,19 +500,27 @@ impl PeerManagerActor {
             total_msg_received_count, max_max_record_num_messages_in_progress, "Bandwidth stats"
         );
 
-        near_performance_metrics::actix::run_later(ctx, every, move |act, ctx| {
-            act.report_bandwidth_stats_trigger(ctx, every);
-        });
+        near_performance_metrics::actix::run_later(
+            ctx,
+            every.try_into().unwrap(),
+            move |act, ctx| {
+                act.report_bandwidth_stats_trigger(ctx, every);
+            },
+        );
     }
 
     /// Receives list of edges that were verified, in a trigger every 20ms, and adds them to
     /// the routing table.
-    fn broadcast_validated_edges_trigger(&mut self, ctx: &mut Context<Self>, interval: Duration) {
-        let start = Clock::instant();
+    fn broadcast_validated_edges_trigger(
+        &mut self,
+        ctx: &mut Context<Self>,
+        interval: time::Duration,
+    ) {
+        let start = self.clock.now();
         let mut new_edges = Vec::new();
         while let Some(edge) = self.routing_table_exchange_helper.edges_to_add_receiver.pop() {
             new_edges.push(edge);
-            if start.elapsed() >= BROAD_CAST_EDGES_MAX_WORK_ALLOWED {
+            if self.clock.now() - start >= BROAD_CAST_EDGES_MAX_WORK_ALLOWED {
```

Comment:
nit: MAX_WORK_ALLOWED is a strange variable for something that is in milliseconds..

Reply:
something that SHOULD be in milliseconds :P. Seriously though, I don't know why we have this check here. Added a TODO to investigate that.